libdevice/imf_wrapper.cpp

```diff
@@ -0,0 +1,162 @@
+//==----- imf_wrapper.cpp - wrappers for intel math library functions ------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "device_imf.hpp"
+
+#ifdef __LIBDEVICE_IMF_ENABLED__
+
+DEVICE_EXTERN_C_INLINE
+int __devicelib_imf_float2int_rd(float x);
+
+DEVICE_EXTERN_C_INLINE
+int __devicelib_imf_float2int_rn(float x);
+
+DEVICE_EXTERN_C_INLINE
+int __devicelib_imf_float2int_ru(float x);
+
+DEVICE_EXTERN_C_INLINE
+int __devicelib_imf_float2int_rz(float x);
+
+DEVICE_EXTERN_C_INLINE
+unsigned int __devicelib_imf_float2uint_rd(float x);
+
+DEVICE_EXTERN_C_INLINE
+unsigned int __devicelib_imf_float2uint_rn(float x);
+
+DEVICE_EXTERN_C_INLINE
+unsigned int __devicelib_imf_float2uint_ru(float x);
+
+DEVICE_EXTERN_C_INLINE
+unsigned int __devicelib_imf_float2uint_rz(float x);
+
+DEVICE_EXTERN_C_INLINE
+long long int __devicelib_imf_float2ll_rd(float x);
+
+DEVICE_EXTERN_C_INLINE
+long long int __devicelib_imf_float2ll_rn(float x);
+
+DEVICE_EXTERN_C_INLINE
+long long int __devicelib_imf_float2ll_ru(float x);
+
+DEVICE_EXTERN_C_INLINE
+long long int __devicelib_imf_float2ll_rz(float x);
+
+DEVICE_EXTERN_C_INLINE
+unsigned long long int __devicelib_imf_float2ull_rd(float x);
+
+DEVICE_EXTERN_C_INLINE
+unsigned long long int __devicelib_imf_float2ull_rn(float x);
+
+DEVICE_EXTERN_C_INLINE
+unsigned long long int __devicelib_imf_float2ull_ru(float x);
+
+DEVICE_EXTERN_C_INLINE
+unsigned long long int __devicelib_imf_float2ull_rz(float x);
+
+DEVICE_EXTERN_C_INLINE
+int __devicelib_imf_float_as_int(float x);
+
+DEVICE_EXTERN_C_INLINE
+unsigned int __devicelib_imf_float_as_uint(float x);
+
+DEVICE_EXTERN_C_INLINE
+float __imf_saturatef(float x) { return __devicelib_imf_saturatef(x); }
+
+DEVICE_EXTERN_C_INLINE
+float __imf_expf(float x) { return __devicelib_imf_expf(x); }
```

Comment:
I suppose that this list is going to be expanded as new math functions added under that ./imf folder. Does it make sense to also move it under the ./imf folder in a separate file and give this file ownership to Numerics team? Otherwise we would need to contact Compiler team after every new function addition in order to expand this list.